worldedit-core/src/main/java/com/sk89q/worldedit/math/convolution/SnowHeightMap.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * WorldEdit, a Minecraft world manipulation toolkit
+ * Copyright (C) sk89q <http://www.sk89q.com>
+ * Copyright (C) WorldEdit team and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.sk89q.worldedit.math.convolution;
+
+import com.sk89q.worldedit.EditSession;
+import com.sk89q.worldedit.MaxChangedBlocksException;
+import com.sk89q.worldedit.function.mask.Mask;
+import com.sk89q.worldedit.math.BlockVector3;
+import com.sk89q.worldedit.regions.Region;
+import com.sk89q.worldedit.registry.state.Property;
+import com.sk89q.worldedit.world.block.BlockState;
+import com.sk89q.worldedit.world.block.BlockTypes;
+
+import javax.annotation.Nullable;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Allows applications of Kernels onto the region's height map for snow layers
+ *
+ * <p>Currently only used for snow layer smoothing (with a GaussianKernel)</p>.
+ */
+public class SnowHeightMap {
+    private final float[] data;
+    private final int width;
+    private final int height;
+
+    private final Region region;
+    private final EditSession session;
+
+    private final Property<Object> layers;
```

Comment:
You should store this as Property<Integer> and cast when getting the property, rather than casting on the getters and setters

Reply:
I'm a little bit confused. I should store the Property as Propert<?>? But why?
e.g. I found [this class](https://github.com/EngineHub/WorldEdit/blob/master/worldedit-core/src/main/java/com/sk89q/worldedit/extent/clipboard/io/legacycompat/BannerBlockCompatibilityHandler.java#L40) using Properties and there is the cast on the getter. I would change the Property<Object> to Property<Integer> and add a try-catch.

Reply:
can probably be stored as an IntegerProperty directly, we're not very consistent throughout the project on using this correctly lol.